test/unit/controllers/EmailConfirmedControllerSpec.scala

```diff
@@ -16,21 +16,59 @@
 
 package unit.controllers
 
+import org.mockito.ArgumentMatchers.{eq => meq, _}
+import org.mockito.Mockito.{times, verify, _}
+import org.scalatest.BeforeAndAfterEach
 import play.api.test.Helpers.{status, _}
+import uk.gov.hmrc.auth.core.EnrolmentIdentifier
 import uk.gov.hmrc.customs.emailfrontend.controllers.EmailConfirmedController
+import uk.gov.hmrc.customs.emailfrontend.model.EmailStatus
+import uk.gov.hmrc.customs.emailfrontend.services.{CustomsDataStoreService, EmailCacheService}
 import uk.gov.hmrc.customs.emailfrontend.views.html.email_confirmed
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
 
+import scala.concurrent.Future
 
-class EmailConfirmedControllerSpec extends ControllerSpec {
+
+class EmailConfirmedControllerSpec extends ControllerSpec with BeforeAndAfterEach {
 
   private val view = app.injector.instanceOf[email_confirmed]
-  private val controller = new EmailConfirmedController(fakeAction, view)
+  private val mockEmailCacheService = mock[EmailCacheService]
+  private val mockCustomsDataStoreService = mock[CustomsDataStoreService]
+
+  private val controller = new EmailConfirmedController(fakeAction, view, mockCustomsDataStoreService, mockEmailCacheService, mcc)
+
+  override protected def beforeEach(): Unit = {
+    reset(mockCustomsDataStoreService)
+  }
 
   "EmailConfirmedController" should {
+    "have a status of OK when email found in cache" in withAuthorisedUser() {
+      when(mockEmailCacheService.fetchEmail(any())(any(), any())).thenReturn(Future.successful(Some(EmailStatus("abc@def.com"))))
+      when(mockCustomsDataStoreService.storeEmail(any(), any())(any())).thenReturn(Future.successful(HttpResponse(OK)))
+
+      val eventualResult = controller.show(request)
+      status(eventualResult) shouldBe OK
+
+      verify(mockCustomsDataStoreService).storeEmail(meq(EnrolmentIdentifier("EORINumber", "GB1234567890")), meq("abc@def.com"))(any[HeaderCarrier])
```

Comment:
why don't we train our mocks to verify the values for params and avoid writing this verify call here?